main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -143,6 +143,37 @@ object WeederError {
 
   }
 
+  /**
+    * An error raised to indicate a struct contains duplicate fields
+    *
+    * @param loc the location of the struct declaration
+    * @param fieldName the name of the field
+    * @param structName the name of the struct
+    * @param field1 the location of the first field
+    * @param field2 the location of the second field
+    */
+  case class DuplicateStructField(loc: SourceLocation, fieldName: String, structName: String, field1: SourceLocation, field2: SourceLocation) extends WeederError with Recoverable {
```

Comment:
I would like the following order of fields:

structName
fieldName
field1Loc
field2Loc
loc

This matches how we do it in most other error messages.